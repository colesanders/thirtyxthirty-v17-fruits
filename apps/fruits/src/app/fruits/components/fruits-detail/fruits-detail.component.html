<mat-card class="mat-card-light"> 
    <mat-card-header style="display: flex; justify-content: space-between;">
      <mat-card-title class="accent">
        <span *ngIf="fruitForm.value.id !== null; else elseBlock">
          Editing Fruit
        </span>
        <ng-template #elseBlock>
          Create Fruit
        </ng-template>
      </mat-card-title>
      
    </mat-card-header>
  <form [formGroup]="fruitForm" (ngSubmit)="saved.emit(fruitForm.value)">
    <mat-card-content>

      <mat-form-field>
        <mat-label>Name</mat-label>
        <input matInput formControlName="name" type="string">
      </mat-form-field>
      <br>


      <mat-form-field>
        <mat-label>Description</mat-label>
        <input matInput formControlName="description" type="string">
      </mat-form-field>
      <br>

      <mat-form-field>
        <mat-label [style]="'color: ' + fruitForm.value.color">Color</mat-label>
        <input matInput formControlName="color" type="string">
      </mat-form-field>
      <br>


      <mat-form-field>
        <mat-label>Amount</mat-label>
        <input matInput formControlName="amount" type="number">
      </mat-form-field>
      <br>

      <section>
        <mat-button-toggle-group style="width:100%;" formControlName="icon">
          <mat-button-toggle style="width: 25%" *ngFor="let icon of fruitIcons" [value]="icon">
            <mat-icon [svgIcon]="icon" ></mat-icon>
          </mat-button-toggle>
        </mat-button-toggle-group>
      </section>
      <br>

    </mat-card-content>
    <mat-card-actions style="display: flex; justify-content: space-between;">
      <button [disabled]="fruitForm.invalid" type="submit" mat-button color="primary">Save</button>
      <button type="button" mat-button (click)="cancelled.emit();$event.stopImmediatePropagation();cancel();">
        Cancel
      </button>
    </mat-card-actions>
  </form>
</mat-card>