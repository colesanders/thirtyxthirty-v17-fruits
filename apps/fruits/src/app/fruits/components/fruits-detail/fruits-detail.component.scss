@import '~@angular/material/theming';

mat-slider {
    width: 100%;
    margin: auto;
}

mat-form-field {
    width: 100%;
}

.star-color {
    color: gold;
}

h4 {
    text-align: center; 
    color: mediumblue;
}

mat-radio-button {
    padding-right: 5px;
}