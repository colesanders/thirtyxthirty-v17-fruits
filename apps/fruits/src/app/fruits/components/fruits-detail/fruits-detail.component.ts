import { Component, OnInit, Input, Output, EventEmitter, OnChanges } from '@angular/core';
import {FormGroup, FormControl, FormBuilder, Validators } from '@angular/forms';

import { Fruit } from '@thirty/api-interfaces';


@Component({
  selector: 'thirty-fruits-detail',
  templateUrl: './fruits-detail.component.html',
  styleUrls: ['./fruits-detail.component.scss']
})
export class FruitsDetailComponent implements OnInit, OnChanges{
  @Input() fruit: Fruit;
  @Output() saved = new EventEmitter();
  @Output() cancelled = new EventEmitter();

  fruitForm: FormGroup;

  fruitIcons = ['thirty-apple', 'thirty-banana', 'thirty-orange', 'thirty-passion-fruit'];

  constructor(private formBuilder: FormBuilder) { }

  ngOnInit(): void {
    this.createFormGroup();
  }

  ngOnChanges(){
    if(this.fruitForm && this.fruit){
      this.fruitForm.patchValue(this.fruit)
    } else if(this.fruitForm){
      this.cancel();
    }
  }

  cancel(){
    this.fruitForm.reset();
    this.fruitForm.value.weight = 0;
  }

  createFormGroup(){
    this.fruitForm = this.formBuilder.group({
      id: [],
      name: new FormControl('', [
        Validators.required,
      ]),
      description: new FormControl('', [
        Validators.required,
      ]),
      color: new FormControl('', [
      ]),
      amount: new FormControl('', [
        Validators.required,
      ]),
      icon: new FormControl('', [
        Validators.required,
      ])
    })
  }
}
