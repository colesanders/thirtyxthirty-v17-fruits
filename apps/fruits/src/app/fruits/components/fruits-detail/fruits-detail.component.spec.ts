import { HttpClientModule } from '@angular/common/http';
import { DebugElement } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { MaterialModule } from '@thirty/material';

import { FruitsDetailComponent } from './fruits-detail.component';

describe('FruitsDetailComponent', () => {
  let component: FruitsDetailComponent;
  let fixture: ComponentFixture<FruitsDetailComponent>;
  let de: DebugElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ FruitsDetailComponent ],
      imports: [
        ReactiveFormsModule,
        HttpClientModule,
        MaterialModule,
        NoopAnimationsModule,
        RouterTestingModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(FruitsDetailComponent);
    component = fixture.componentInstance;
    de = fixture.debugElement;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should create the formGroup', () => {
    component.createFormGroup();
    expect(component.fruitForm).toBeTruthy();
  });

  it('should reset formGroup', () => {
    component.cancel();
    expect(component.fruitForm.value).toMatchSnapshot();
  });


});
