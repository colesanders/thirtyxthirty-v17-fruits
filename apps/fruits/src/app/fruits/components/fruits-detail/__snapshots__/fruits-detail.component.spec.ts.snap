// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FruitsDetailComponent should reset formGroup 1`] = `
Object {
  "id": null,
  "name": null,
  "species": null,
  "typing": null,
  "weight": 0,
}
`;
